Man in the middle

appasalvi opened the pull request:
Hi Michael and Aaron,

There are some issues with this commit (e.g. we get 4 extra tabs when we execute this code).
 The icons are working fine (i.e. they turn red when we go to websites like serebii.net).

Sorry I didn't commit earlier but I wanted to make some more improvements first to the requestinterceptor and manifest files.

Thanks.
Satya

appasalvi committed:
HelloWorldExample

appasalvi committed:
Merge branch 'InitialDev' of https://github.com/Cyberspace-Citizenship-Coefficient/Chrome-Plugin into InitialDev

appasalvi committed:
Start :smile:

appasalvi committed:
Merge branch 'InitialDev' of https://github.com/Cyberspace-Citizenship-Coefficient/Chrome-Plugin into InitialDev

appasalvi committed:
Merge branch 'InitialDev' into ManInTheMiddle

appasalvi committed:
Developing Man in the Middle but there are some issues like we get 4 extra tabs opened in the browser.

TheENGR commented on src/manifest.json:
```diff
@@ -13,7 +13,11 @@
+		"scripting"
+	],
+	"host_permissions":[
+		"https://www.serebii.net/"
```
Probs need more

TheENGR replied:
```suggestion
		"https://*/"
```

TheENGR commented on src/manifest.chrome.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Cyber Citizenship Coefficient",
+  "description": "Build an Chrome Plugin Extension",
```
Do we need this at all?

appasalvi replied:
Hi TheENGR,

I deleted the manifest.chrome.json file.  

Thanks.
Satya

TheENGR replied:
You need to push that deletion

TheENGR commented on src/siteQualityMapping.js:
```diff
@@ -46,10 +46,12 @@ chrome.contextMenus.onClicked.addListener(async (info, tab) => {
 
 
 chrome.tabs.onUpdated.addListener(async (tabId, changeInfo, tab) => {
+	console.log('onUpdate');
```
Don’t push debugging logs

appasalvi replied:
Hi TheENGR,

This has been resolved as I have removed the console.log lines.

TheENGR commented on src/siteQualityMapping.js:
```diff
@@ -46,10 +46,12 @@ chrome.contextMenus.onClicked.addListener(async (info, tab) => {
 })
 
 chrome.tabs.onActivated.addListener(async (activeInfo) => {
+	console.log('onActivated');
```
Remove debugging logs

TheENGR commented on src/requestinterceptor.js:
```diff
@@ -9,4 +9,22 @@ chrome.tabs.onUpdated.addListener(async (tabId, changeInfo, tab) => {
+    await chrome.storage.local.get(["blockedSites", "warnedSites"], async (storage) => {
+        if (storage.warnedSites.find(x => URL.includes(x)) || storage.blockedSites.find(x => URL.includes(x))) {
+            // close target
+            chrome.tabs.remove(tabId)
```
Why not just update the tab to have a new URL?

https://developer.chrome.com/docs/extensions/reference/tabs/#method-update

appasalvi replied:
Hi TheENGR,

What is the advantage of not uploading a new tab?  If the "Man in the Middle" is created within the existing tab then we could run into issues with the browser history.

TheENGR replied:
We MIGHT be able to re-write the history

Wanted to not load jerks so they don't get paid for ADs

TheENGR replied:
https://developer.chrome.com/docs/extensions/reference/history/ 

We can re-write the history if needed

TheENGR commented:
Definitely solid forward progress, but let’s not merge it just yet :)

appasalvi committed:
Update src/manifest.json

Co-authored-by: TheENGR <50523916+TheENGR@users.noreply.github.com>

appasalvi committed:
Story:
1. Removed debugging logs
2. Prevented multiple tabs from opening upon firing blocked/warned sites
3. Changed Host Permissions in manifest.json file

appasalvi committed:
Story:

appasalvi committed:
Added Blocked Sites, Warned Sites and Good Sites

appasalvi committed:
Update README.md

appasalvi committed:
Added two buttons for "Go" and "Always Go", and now we hvae created all 3 buttons!

appasalvi committed:
Merge branch 'InitialDev' into ManInTheMiddle

appasalvi committed:
WIP

appasalvi committed:
Merge branch 'ManInTheMiddle' of https://github.com/Cyberspace-Citizenship-Coefficient/Chrome-Plugin into ManInTheMiddle

appasalvi committed:
WIP 2

appasalvi committed:
WIP_2

appasalvi committed:
1. Buttons are nearly all functional. We have stored the whitelist, redlist, and yellowlist.  However I'm not sure what the greenlist would be for exactly.   In order to do the local cache storage we need to get the lambda to accept the request.  Finally I don't believe we need to clear local cache as once we close a Chrome browser and open it up again the local cache will be cleared.

appasalvi committed:
All the buttons are working now and I'm working on the remainder of the To Do items which include the local cache storage tasks.

TheENGR commented on src/contentScript.js:
```diff
@@ -1,13 +1,14 @@
 document.addEventListener('contextmenu', e => {
 	// Send messages to ourself
-	chrome.runtime.sendMessage("afffkcmpebnikjnoagiiofainbpffnch", {
+	chrome.runtime.sendMessage("GoingBack", {
```
That string should have to be the app code, as it was

TheENGR commented on src/display.js:
```diff
@@ -0,0 +1,44 @@
+    chrome.runtime.sendMessage("afffkcmpebnikjnoagiiofainbpffnch",
+    {
+        action: "always go",
+        url: findParam('url')+"?skip_interceptor" //setting flag so that the interceptor ignores this request
```
You can't just add a `?`, there probably is one on most sites already 
https://www.wsj.com/articles/miami-condo-buildings-old-age-collapse-champlain-towers-11626271508?mod=hp_lead_pos5
See

appasalvi committed:
Nearly all action items on the To Do List are completed and I need Aaron's help to determine the public endpoint for Chrome

TheENGR reviewed (approved):
Solid progress! 

Time for a handoff 🔋
